protocol/downloader.go

```diff
@@ -91,46 +100,62 @@ func (d *Downloader) SyncBlockchain(forkResolver ForkResolver) {
 			d.log.Info(fmt.Sprintf("Node is synchronized"))
 			return
 		}
-		d.batches = make(chan *batch, 10)
-		term := make(chan interface{})
-		completed := make(chan interface{})
-		go d.consumeBlocks(term, completed)
-
-		from := head.Height() + 1
-	loop:
-		for from <= d.top {
-			for peer, height := range knownHeights {
-				if height < from {
-					continue
-				}
-				to := math.Min(from+BatchSize, height)
-				if err, batch := d.pm.GetBlocksRange(peer, from, to); err != nil {
-					continue
-				} else {
-					select {
-					case d.batches <- batch:
-					case <-term:
-						break loop
-					}
+		if !d.isSyncing {
```

Comment:
When program starts, d.isSyncing will be `true`, since that, it will always be true?